### ISSUE TITLE
Make obscure "no signature of method" error message useful

### ISSUE THREAD

**Issue by ewirch** [gradle/gradle#19282]
### Expected Behavior
An error message like:
```
A problem occurred evaluating project ':api'.
> Unresolved method 'testFixtures()' in api/build.gradle:73
```

### Current Behavior
Error message looks like this:
```
A problem occurred evaluating project ':api'.
> No signature of method: build_do4ag1ea149cg39mn6p1dcdkq.testing() is applicable for argument types: (build_do4ag1ea149cg39mn6p1dcdkq$_run_closure5) values: [build_do4ag1ea149cg39mn6p1dcdkq$_run_closure5@59d02712]
  Possible solutions: toString(), toString(), toString(), toString()
```

### Context
The message is triggered by `testFixtures` in:
```
testing {
    suites {
        test {
            useJUnitJupiter()

            sources {
                java {
                    srcDirs = ['src/test/java']
                }
            }
        }

        transactionalIntegrationTest(JvmTestSuite) {
            useJUnitJupiter()

            dependencies {
                implementation project

                implementation testFixtures(project(':common'))
            }

            targets {
                all {
                    testTask.configure {
                        shouldRunAfter(test)
                    }
                }
            }

            sources {
                java {
                    srcDirs = ['src/it/transactional/java']
                }
            }
        }
    }
}
```

**Comment by melix** [comment]
So this problem is driving a lot of people nuts. I'm aware of the problem so usually, when I see an error message like:

```
         > No signature of method: build_bdowaa3v2zbcd3yeq18rn1b51.graalvmNative() is applicable for argument types: (build_bdowaa3v2zbcd3yeq18rn1b51$_run_closure4) values: [build_bdowaa3v2zbcd3yeq18rn1b51$_run_closure4@43ce4772]
```

and a build file with:

```
graalvmNative {
    binaries {
        main {
             buildArgs << "foo"
        }
    }
}
```

I have enough expertise to understand that the problem is _not_ with `graalvmNative { ... }` not being a method, but that the bug is actually _within_ the block (can you spot it? `buildArgs << "foo"` should be `buildArgs.add("foo")`. The only way to figure out what line within the block is faulty is to comment out things one by one.

I don't know when Gradle started to report the error on the containing block instead of the root cause, but this is extremely user unfriendly.

**Comment by ljacomet** [comment]
Fix will be release with Gradle 7.5

**Comment by AleXoTroN** [comment]
For errors like this you can also ask chatgpt lol. Nothing helped and then i asked it xD
![grafik](https://user-images.githubusercontent.com/13837466/220196451-0b925f2a-dbe5-4467-b6fd-8dadcc2a5086.png)
